test/tracing/mssql/app.js

```diff
@@ -0,0 +1,552 @@
+/* eslint-disable no-console */
+
+'use strict';
+
+require('../../../')({
+  agentPort: process.env.AGENT_PORT,
+  level: 'info',
+  tracing: {
+    forceTransmissionStartingAt: 1
+  }
+});
+
+
+var bodyParser = require('body-parser');
+var express = require('express');
+var morgan = require('morgan');
+var sql = require('mssql');
+var devNull = require('dev-null');
+
+var pool;
+var app = express();
+var logPrefix = 'Express / MSSQL App (' + process.pid + '):\t';
+
+
+sql.on('error', function(err) {
+  log(err);
+});
+
+
+var dbHost = process.env.MSSQL_HOST ? process.env.MSSQL_HOST : '127.0.0.1';
+var dbPort = process.env.MSSQL_PORT ? parseInt(process.env.MSSQL_PORT, 10) : 1433;
+var dbUrl = dbHost + ':' + dbPort;
+var dbUser = process.env.MSSQL_USER ? process.env.MSSQL_USER : 'sa';
+var dbPassword = process.env.MSSQL_PW ? process.env.MSSQL_PW : 'stanCanHazMsSQL1';
+var initConnectString = 'mssql://' + dbUser + ':' + dbPassword + '@' + dbUrl + '/tempdb';
+var dbName = 'nodejssensor';
+var ready = false;
+
+
+sql
+  .connect(initConnectString)
+  .then(function() {
+    return new sql.Request().query(
+      'IF EXISTS (SELECT * FROM sys.databases WHERE name = N\'' + dbName + '\') DROP DATABASE ' + dbName
+    );
+  })
+  .then(function() {
+    return new sql.Request().query('CREATE DATABASE ' + dbName);
+  })
+  .then(function() {
+    return sql.close();
+  })
+  .then(function() {
+    return sql.connect({
+      user: dbUser,
+      password: dbPassword,
+      server: dbHost,
+      port: dbPort,
+      database: dbName
+    });
+  })
+  .then(function(_pool) {
+    pool = _pool;
+    return new sql.Request().query(
+      'CREATE TABLE UserTable (id INT IDENTITY(1,1), name VARCHAR(40) NOT NULL, email VARCHAR(40) NOT NULL)'
+    );
+  })
+  .then(function() {
+    return new sql.Request().batch(
+      'CREATE PROCEDURE testProcedure' +
+      '    @username nvarchar(40)' +
+      'AS' +
+      '    SET NOCOUNT ON;' +
+      '    SELECT name, email' +
+      '    FROM UserTable' +
+      '    WHERE name = @username;'
+    );
+  })
+  .then(function() {
```

Comment:
Can we extend this app setup with the creation of the prepared statement that is then used within one of the HTTP handlers? That way we could verify the comment I made above.

Reply:
Funny, only saw this comment now but that's exactly what I did in the last commit.

And turns you were somewhat right: cls.ns.set actually throws in that setup because there hasn't been any previous call to cls.ns.bind et al.